evm/auxiliary/user_account.py

```diff
@@ -0,0 +1,181 @@
+from eth_keys import keys
+
+from evm.vm.forks.sharding.transactions import (
+    ShardingTransaction,
+)
+
+from evm.validation import (
+    validate_uint256,
+    validate_canonical_address,
+    validate_is_bytes,
+)
+
+from evm.utils.hexadecimal import (
+    decode_hex,
+)
+from evm.utils.numeric import (
+    big_endian_to_int,
+    int_to_big_endian,
+)
+from evm.utils.padding import (
+    pad32,
+)
+
+
+USER_ACCOUNT_CODE = decode_hex(
+    "0x3f595260805903608020600052602060006080600060006001610bb8f1506001600051"
+    "1461002c57600080fd5b6000546060351461003c57600080fd5b60016000540160005560"
+    "8035f55060006000610120590361010060a03560c0356113885a03f13d60003e50801561"
+    "0076573d6000f361007b565b3d6000f35b50"
+)
+
+
+SIGNATURE_V_SLICE = slice(0, 1 * 32)
+SIGNATURE_R_SLICE = slice(1 * 32, 2 * 32)
+SIGNATURE_S_SLICE = slice(2 * 32, 3 * 32)
+SIGNATURE_SLICE = slice(SIGNATURE_V_SLICE.start, SIGNATURE_S_SLICE.stop)
+NONCE_SLICE = slice(3 * 32, 4 * 32)
+GASPRICE_SLICE = slice(4 * 32, 5 * 32)
+VALUE_SLICE = slice(5 * 32, 6 * 32)
+RECEIVER_SLICE = slice(6 * 32, 7 * 32)
+DATA_SLICE = slice(7 * 32, None)  # open ended
+
+EMPTY_DATA = b'\x00' * RECEIVER_SLICE.stop
+
+
+class StandardTransaction(ShardingTransaction):
+    """A transaction calling the standard user account contract.
+
+    The contract expects transaction data to be formatted as:
+        # 0-95: sig
+        # 96-127: nonce
+        # 128-159: gasprice
+        # 160-191: value
+        # 192-223: to
+        # 224+: data
+    """
+
+    def __init__(
+        self,
+        chain_id,
+        shard_id,
+        target,
+        start_gas,
+        access_list,
+        receiver,
+        value,
+        nonce,
+        gas_price,
+        msg_data
+    ):
+        super().__init__(
+            chain_id=chain_id,
+            shard_id=shard_id,
+            target=target,
+            data=EMPTY_DATA,
+            start_gas=start_gas,
+            access_list=access_list,
+            code=b'',
+        )
+        self.receiver = receiver
+        self.value = value
+        self.nonce = nonce
+        self.gas_price = gas_price
+        self.msg_data = msg_data
+
+    #
+    # Properties for fields encoded in data
+    #
+
+    @property
+    def signature(self):
```

Comment:
what do you think about renaming this to `vrs` instead of `signature`?  I think that naming convention would match better with other portions of both this codebase and others (web3.py and eth-keys).

With the conventions we've been establishing, I would expect this to return a `eth_keys.Signature` object.

Reply:
Do you prefer renaming it to `vrs` or making it return a `Signature` object? I have no preference in that regard.

Reply:
I think `vrs` is my preference since this is a low-level API